app/auth/forms.py

```diff
@@ -47,8 +48,14 @@ class RegistrationForm(FlaskForm):
     password = PasswordField('Password', validators=[
         DataRequired(), 
         EqualTo('password2', message='Passwords must match.'),
-        Length(8,64, message='Password must be at least 8 characters in length.'),
-        PasswordStrength()
+        Length(min=8, message='Password must be at least 8 characters in length.'),
+        PasswordStrength(),
+        Regexp(
+            '^(?=.*[0-9].*[0-9]).*$',
+            0, 'Passwords must contain at least two digits.'),
+        Regexp(
+            '^(?=.*[A-Z].*[A-Z]).*$',
+            0, 'Passwords must contain at least two uppercase letters.')
```

Comment:
Requiring 2 caps and digits seems overzealous compared to norms and will frustrate users, I think. It's also not really a good direction, considering the NCSC advice:

> For this reason, enforcing the requirement for complex character sets in passwords is not recommended. Instead, concentrate efforts on technical controls

https://www.ncsc.gov.uk/guidance/password-guidance-simplifying-your-approach

Maybe just require one of each.